lib/game.rb

```diff
@@ -0,0 +1,66 @@
+# frozen_string_literal: true
+
+require_relative 'board.rb'
+require_relative 'player.rb'
+
+class Game
+  def initialize(player_one, player_two, board)
+    @player_one = player_one
+    @player_two = player_two
+    @board = board
+  end
+
+  def switch_player(active_player)
+    if active_player == @player_one
+      active_player = @player_two
+    elsif active_player == @player_two
+      active_player = @player_one
+    end
+    active_player
+  end
+
+  def check_rows(game_over)
+    @board.sq_val.each_with_index do |_value, i|
+      next if i % 3 != 0
+
+      if @board.sq_val[i] != ' '
+        game_over = true if @board.sq_val[i] == @board.sq_val[i + 1] && @board.sq_val[i] == @board.sq_val[i + 2]
+      end
+    end
+    game_over
+  end
+
+  def check_columns(game_over)
+    @board.sq_val.each_with_index do |_value, i|
+      next if i > 2
+
+      if @board.sq_val[i] != ' '
+        game_over = true if @board.sq_val[i] == @board.sq_val[i + 3] && @board.sq_val[i] == @board.sq_val[i + 6]
+      end
+    end
+    game_over
+  end
+
+  def check_diagonal_one(game_over)
+    if @board.sq_val[0] != ' '
+      game_over = true if @board.sq_val[0] == @board.sq_val[4] && @board.sq_val[0] == @board.sq_val[8]
+    end
+    game_over
+  end
+
+  def check_diagonal_two(game_over)
+    if @board.sq_val[2] != ' '
+      game_over = true if @board.sq_val[2] == @board.sq_val[4] && @board.sq_val[2] == @board.sq_val[6]
+    end
+    game_over
+  end
+
+  def determine_winner(game_over)
+    return true if check_rows(game_over) == true
+    return true if check_columns(game_over) == true
+    return true if check_diagonal_one(game_over) == true
+    return true if check_diagonal_two(game_over) == true
+
+    game_over
+  end
```

Comment:
The current way for checking a winner is too complex, firstly you do not need to search for different winning positions separately, what you can do is to get the possible winning combinations into an array and loop through the array checking for the same value in each subarray.
[[1,2,3],[4,5,6],......]
then loop through this array checking if each subarray has same element, if it does then the last player is the winner

with this logic above method 
check_row
check_column
check_diagonal_one
check_diagonal_two
determine_winner 

can all go away

Reply:
Your suggestion sounds brilliantly complex. 
We tried to combine the checks into one nice method, however, we ran into many stickler problems saying that we were trying to do too much in a single method. That is why we had to break it up into multiple methods.
We don't see how we can combine it again without running into the same stickler problems again.